src/Uno.UI/UI/Xaml/UIElement.wasm.cs

```diff
@@ -175,7 +166,16 @@ protected internal void SetClasses(string[] cssClasses, int index = -1)
 		/// <param name="clipRect">The Clip rect to set, if any</param>
 		protected internal void ArrangeElementNative(Rect rect, bool clipToBounds, Rect? clipRect)
 		{
-			LayoutSlotWithMarginsAndAlignments = rect;
+			LayoutSlotWithMarginsAndAlignments =
+				VisualTreeHelper.GetParent(this) is UIElement parent
+					? rect.DeflateBy(parent.GetBorderThickness())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'rect' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826507)